client/containers/create-update-profile.js

```diff
@@ -5,37 +5,39 @@ import ProfileForm from '../components/profile-form'
 export default class CreateUpdateProfile extends Component {
   constructor(props) {
     super(props)
-    this.createProfile = this.createProfile.bind(this)
-    this.updateProfile = this.updateProfile.bind(this)
+    this.saveProfile = this.saveProfile.bind(this)
   }
-  createProfile(userDetails) {
-    const { updateUser } = this.props
-    const url = '/artists'
+  saveProfile(userDetails) {
+    const { updateUser, navigate, path, user: { displayName } } = this.props
+    const url = path === 'update-profile'
+      ? '/artists/' + displayName
+      : '/artists'
+    const method = path === 'update-profile'
+      ? 'PUT'
+      : 'POST'
+    const navigatePath = path === 'update-profile'
+      ? '#' + displayName
+      : 'add-music'
+    const alertMessage = path === 'update-profile'
+      ? 'Your profile has been updated.'
+      : 'Your profile has been saved.'
     const req = {
-      method: 'POST',
+      method: method,
       body: JSON.stringify(userDetails),
       headers: { 'Content-Type': 'application/json' }
     }
-    request.sendFetch(url, req, updateUser)
-  }
-  updateProfile(userDetails) {
-    const { updateUser, user: { displayName } } = this.props
-    const url = '/artists/' + displayName
-    const req = {
-      method: 'PUT',
-      body: JSON.stringify(userDetails),
-      headers: { 'Content-Type': 'application/json' }
-    }
-    request.sendFetch(url, req, updateUser)
+    request.sendFetch(url, req)
+      .then(user => updateUser(user))
+      .then(alert(alertMessage))
```

Comment:
Don't forget, `then` takes a function _definition_, not a function _call_. This `alert` and `navigate` are not actually waiting for your `sendFetch` to complete.